gemini/sample-apps/accelerating_product_innovation/app/pages_utils/utils_export_content_pdf.py

```diff
@@ -0,0 +1,173 @@
+
+        # Add text content, handling potential page breaks
+        pages = check_add_page(pdf, text)
+        pdf.set_font("Arial", "", 11)
+
+        for i, page in enumerate(pages):
+            if page.strip() == "":
+                continue
+
+            if i == 0:  # First page of text
+                pdf.set_xy(17, 120)
+            else:  # Subsequent pages
+                add_formatted_page(pdf)
+                pdf.set_xy(17, 15)
+                pdf.set_font("Arial", "", 11)
+
+            pdf.multi_cell(170, 5, page)  # Output the text
+
+
+def create_content_pdf(product_content: list, selected_titles: list[str]) -> None:
+    """Creates a PDF for each product content and selected title.
+
+    Args:
+        product_content: A list of lists of dictionaries representing the
+        product content.
+        selected_titles: A list of selected titles for each product content.
+    """
+    print("HERE....")
+    for product_index in range(len(product_content) - 1):
+        pdf = FPDF()  # Create a PDF for the current product
+        content = []
+        images = []
+
+        # Build content and image lists for the current product
+        content.append(product_content[product_index][0]["text"].replace("**", ""))
+        images.append(st.session_state.num_drafts * product_index + 1)
+
+        # Generate the PDF layout
+        create_pdf_layout(pdf, content, selected_titles[product_index], images)
+
+        # Save the PDF with an appropriate filename
+        print("SAVING")
+
+        pdf.output(f"content_{product_index}.pdf")
+
+
+def cut_string(string: str, num_characters: int) -> str:
+    """Cuts a string to the specified number of characters.
+
+    Args:
+        string: The string to cut.
+        num_characters: The number of characters to cut the string to.
+
+    Returns:
+        The cut string.
+    """
+    if len(string) <= num_characters:
+        return string
+    return string[:num_characters]
+
+
+def create_email_pdf(
+    title: str, email_text: str, filename: str, image_name: str
+) -> None:
+    """Creates a PDF document from an email.
+
+    The PDF document contains the email subject, body, and an image.
+    The title of the PDF document is set to the title of the email.
+
+    Args:
+        title: The title of the email.
+        email_text: The body of the email.
+        filename: The name of the PDF file to be created.
+        image_name: The name of the image file to be included in the PDF
+        document.
+    """
+    pdf = FPDF()
+    parts = email_text.split("\n", 1)
+    subject = parts[0]
+    text = parts[1]
+    add_formatted_page(pdf)
+
+    pdf.set_xy(15, 15)
+    pdf.set_text_color(106, 144, 226)
+    pdf.set_font("Arial", "B", 11)
+    pdf.multi_cell(
+        180,
+        5,
+        f"{title} {st.session_state.product_category}",
+        0,
+        align="C",
+    )
+
+    pdf.set_text_color(0, 0, 0)
+
+    pdf.set_font("Arial", "B", 11)
+    pdf.set_xy(17, 25)
```

Comment:
Can these be set when initalizing the pdf object?

Reply:
We keep changing the text styling and color for heading, then for sub-heading and then text. So, initializing the formatting with pdf object does not seem to be useful. Please unresolve this if you feel otherwise.